### PR TITLE
Improve permalinks docs

### PR DESCRIPTION
**Pull request by tomjohnson1492**
See https://github.com/jekyll/jekyll/pull/5630 for more details on the update. 

@jekyll/documentation

### CODE REVIEW

File: docs/_docs/permalinks.md [hunk 1]
```diff
@@ -4,27 +4,62 @@ title: Permalinks
...
 <div class="note info">
-  <h5>Specifying permalinks through the YAML Front Matter</h5>
-  <p>
-    Built-in permalink styles are not recognized in YAML Front Matter. So
-    <code>permalink: pretty</code> will not work, but the equivalent 
-    <code>/:categories/:year/:month/:day/:title/</code>
-    using template variables will.
-  </p>
+<h5>Specifying permalinks through the YAML Front Matter</h5>
+<p>Built-in permalink styles are not recognized in YAML Front Matter. As a result, <code>permalink: pretty</code> will not work, but the equivalent <code>/:categories/:year/:month/:day/:title/</code> using template variables will.</p>
```
Comment (parkr) [inline docs/_docs/permalinks.md]: This just means you can't use template variables in the `permalink` attribute in the individual files. I think it is mostly true.

File: docs/_docs/permalinks.md [hunk 2]
```diff
@@ -4,27 +4,62 @@ title: Permalinks
...
 </div>
+{% endcomment %}
+
+When you use permalinks that omit the `.html` file extension (called "clean URLs") Jekyll builds the file as index.html placed inside a folder with the page's name. For example:
```
Comment (parkr) [inline docs/_docs/permalinks.md]: We should make the distinction here between "clean URLs" and "pretty URLs", as they can be easily confused. I call these extensionless URL's, as they omit the extension of the file.

File: docs/_docs/permalinks.md [hunk 3]
```diff
@@ -4,27 +4,62 @@ title: Permalinks
...
+
+```
+├── mypageurl
+│   └── index.html
+```
+
+Servers automatically load the index.html file inside of any folder, so users can simply navigate to `http://somedomain.com/mypageurl` to get to `mypageurl/index.html`.
```
Comment (parkr) [inline docs/_docs/permalinks.md]: This is incorrect.

`http://somedomain.com/mypageurl` attempts to read `http://somedomain.com/mypageurl.html`.

`http://somedomain.com/mypageurl/` attempts to read `http://somedomain.com/mypageurl/index.html`.

File: docs/_docs/permalinks.md [hunk 4]
```diff
@@ -280,38 +301,71 @@ Given a post named: `/2009-04-29-slap-chop.md`
...
 
-The Apache web server has very extensive support for content negotiation and can
-handle extensionless URLs by setting the [multiviews][] option in your
-`httpd.conf` or `.htaccess` file:
+The path to the post or page in the built site differs for posts, pages, and collections:
 
-[multiviews]: https://httpd.apache.org/docs/current/content-negotiation.html#multiviews
+### Posts
+
+No matter how many subfolders you organize your posts into inside the `_posts` folder, all posts are pulled out of those subfolders and flattened into the `_site`'s root directory upon build.
```
Comment (parkr) [inline docs/_docs/permalinks.md]: The last bit here is kind of misleading. Could we say:

> The subfolders you may organize your posts into inside `_posts` will not be a part of the permalink.

File: docs/_docs/permalinks.md [hunk 5]
```diff
@@ -280,38 +301,71 @@ Given a post named: `/2009-04-29-slap-chop.md`
...
+
+If you use a permalink style that omits the `.html` file extension, each post is rendered as an `index.html` file inside a folder with the post's name (for example, `categoryname/2016/12/01/mypostname/index.html`).
+
+### Pages
+
+Unlike posts, pages are *not* removed from their subfolder directories when you build your site. Pages remain in the same folder structure in which you organized your pages in the source directory, except that the structure is now mirrored in `_site`. (The only exception is if your page has a `permalink` declared its front matter &mdash; in that case, the structure honors the permalink setting instead of the source folder structure.)
```
Comment (parkr) [inline docs/_docs/permalinks.md]: I don't like talking about what pages *don't* do.

> Unlike posts, pages by default mimic exactly the source directory structure.

File: docs/_docs/permalinks.md (hunk 2 above)
Comment (tomjohnson1492) [inline docs/_docs/permalinks.md]: Actually, in the "Extensionless permalinks" section, it defines these by saying "Jekyll supports permalinks that contain neither a trailing slash nor a file extension." In other words, here is an extensionless permalink by that definition: `/somefileurl`. Is that section wrong? I don't see how "Extensionless" describes a URL that lacks a trailing slash. 

Re clean versus pretty, good point. I didn't realize the difference, but now I see that clean URLs omit the query string, whereas pretty URLs omit the file extension.

File: docs/_docs/permalinks.md (hunk 3 above)
Comment (tomjohnson1492) [inline docs/_docs/permalinks.md]: ahh, i didn't realize this. thanks.

Comment (tomjohnson1492) [inline docs/_docs/permalinks.md]: If I go to `http://jekyllrb.com/news/2016/11/14/jekyll-3-3-1-released`, the server auto-updates the path to `http://jekyllrb.com/news/2016/11/14/jekyll-3-3-1-released/`. Is the server first looking at `http://jekyllrb.com/news/2016/11/14/jekyll-3-3-1-released.html` and then, not finding the file, re-directing to `http://jekyllrb.com/news/2016/11/14/jekyll-3-3-1-released/index.html`?

File: docs/_docs/permalinks.md (hunk 4 above)
Comment (tomjohnson1492) [inline docs/_docs/permalinks.md]: updated

File: docs/_docs/permalinks.md (hunk 5 above)
Comment (tomjohnson1492) [inline docs/_docs/permalinks.md]: okay, updated.

**Comment by tomjohnson1492** [comment]
Made the updates. Submitting for review.

**Comment by tomjohnson1492** [comment]
I have to say I like the care and attention the team places on docs. I wish people were this quick to review docs at my work. One thing I don't get, though, is why the "documentation team" (the Github team organization) seems so quiet.

**Comment by tomjohnson1492** [comment]
Another question. I have the box checked that says "Allow edits from maintainers." This means you can make edits directly on the content. Just wondering why you guys never do that, esp. for small typos or formatting issues. Does it create conflicts or something?

**Comment by DirtyF** [comment]
@jekyllbot: merge +docs

**Pull request closed without merging**